### ISSUE TITLE
🐛 BUG: when using nebula as library in custom application - we cannot handle os.Exit produced by library

### ISSUE THREAD

**Issue by valda-z** [slackhq/nebula#1031]
### What version of `nebula` are you using?

1.7.1

### What operating system are you using?

Linux, macOS, Windows

### Describe the Bug

We are using nebula as library in our own binaries. But because library code contains os.Exit, we cannot handle it.
In our solution we are using customized source code of nebula to handle it correctly.
Problem is with file interface.go on line 285 where is:
			// This only seems to happen when something fatal happens to the fd, so exit.
			os.Exit(2)

We are changing code in our solution to:
         break;

And after break from loop library returns that connection is broken and we can start connecting again.

### Logs from affected hosts

No logs needed

**Comment by wadey** [comment]
I agree this should be fixed. I am curious how this is happening in your system, as I have never seen this os.Exit trigger in the wild myself. Do you have any details on that?

**Comment by johnmaguire** [comment]
Hi @valda-z - I just wanted to bump this ticket again and ask if you have any additional details regarding this line of code being hit? Thanks!

**Comment by valda-z** [comment]
I think that we can see it only in some very rare situations. But unfortunately we have no logs for this because in fact we are restarting our service and it is not issue for us ..
One situation observed is - when network interface are changing - from lan to wifi network card
Second - when disabling networking completely on machine.

**Comment by nbrownus** [comment]
Was able to repro using `ip link delete $deviceName`

**Comment by valda-z** [comment]
Is there any plan when it will be fixed?

**Comment by nbrownus** [comment]
Resolved by #1661
